core/src/main/java/org/pentaho/di/core/plugins/PluginRegistry.java

```diff
@@ -1063,4 +1064,31 @@ public void reset() {
       lock.writeLock().unlock();
     }
   }
+
+  public PluginInterface findPluginWithId( Class<? extends PluginTypeInterface> pluginType, String pluginId, boolean waitForPluginToBeAvailable ) {
+    PluginInterface pluginInterface = findPluginWithId(  pluginType,  pluginId );
+    return waitForPluginToBeAvailable && pluginInterface == null
+      ? waitForPluginToBeAvailable( pluginType,  pluginId, WAIT_FOR_PLUGIN_TO_BE_AVAILABLE_LIMIT )
+      : pluginInterface;
+  }
+
+  private PluginInterface waitForPluginToBeAvailable( Class<? extends PluginTypeInterface> pluginType, String pluginId, int waitLimit ) {
+    int timeToSleep = 50;
+    try {
+      Thread.sleep( timeToSleep );
+      waitLimit -= timeToSleep;
+    } catch ( InterruptedException e ) {
+      log.logError( e.getLocalizedMessage(), e );
+      if ( Thread.currentThread().isInterrupted() ) {
```

Comment:
When Thread.sleep throws interruptedexception, the interrupted status is cleared.  So I would expect the condition on 1082 to always be false.
To reset the interrupt flag, just call `Thread.currentThread().interrupt();`.
That'll allow any callers up the chain in this thread to deal with it.

Good overview here:  https://www.yegor256.com/2015/10/20/interrupted-exception.html